src/quo/components/inputs/address_input/component_spec.cljs

```diff
@@ -142,11 +158,12 @@
         (h/render [address-input/address-input
                    {:on-detect-ens on-detect-ens
                     :ens-regex     ens-regex}])
-        (h/wait-for #(h/is-truthy (h/get-by-label-text :paste-button)))
+        (h/is-truthy (h/query-by-label-text :paste-button))
         (h/fire-event :press (h/get-by-label-text :paste-button))
-        (h/wait-for #(h/is-falsy (h/get-by-label-text :clear-button)))
-        (h/wait-for #(h/is-truthy (h/get-by-label-text :loading-button-container)))
-        (h/was-called on-detect-ens))))
+        (-> (h/wait-for #(h/is-falsy (h/query-by-label-text :clear-button)))
+            (.then (fn []
+                     (h/wait-for #(h/get-by-label-text :loading-button-container))))
+            (.then #(h/was-called on-detect-ens))))))
```

Comment:
FYI: Notice how the next assertion must go after the promise returned by the second `wait-for` is fulfilled.

I think adding the library https://github.com/funcool/promesa to the project is starting to make even more sense to keep the syntax more clean, but I also think these few `.then` calls look just fine.

Reply:
+1 for promesa, I guess this discussion shiould be handled outside of the pr. It would be good to get a clearer understanding for the pros/cons of using that library in the project. 
cc @clauxx

Reply:
Sure, outside this PR. It's like the 5th or so time I remember we kind of wish promesa was in the project. Also even an external contributor mentioned this. I used it a lot in the past, it's kind of a no-brainer.